flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/source/IcebergSource.java

```diff
@@ -464,6 +519,15 @@ public Builder<T> watermarkColumnTimeUnit(TimeUnit timeUnit) {
       return this;
     }
 
+    /**
+     * Optional. Default is no watermark strategy. Only relevant if using the {@link
+     * Builder#buildStream(StreamExecutionEnvironment)}.
+     */
+    public Builder<T> watermarkStrategy(WatermarkStrategy<T> newStrategy) {
```

Comment:
Is there a way to provide a useful `WatermarkStrategy`?
I think it is possible to provide a useful `TimestampAssigner`, but I don't see how can someone provide a useful `WatermarkGenerator`. The only possible way to generate watermarks are with the `watermarkColumn` is provided, but even then the `WatermarkGenerator` should not be used.

Do I miss something?

Reply:
this is only for the `buildStream` method. this just moved the watermark strategy from `env.fromSource` to the builder.

for regular `build` method, users would also need to set the watermark strategy, which most likely would be none.
```
    DataStream<RowData> stream =
        env.fromSource(
            sourceBuilder().build(),
            WatermarkStrategy.noWatermarks(),
            "IcebergSource",
            TypeInformation.of(RowData.class));
```

Reply:
will remove `WatermarkStrategy` from here. we can always add another `buildStream` with a new arg in the future if there is really a need.